collect_app/src/androidTest/java/org/odk/collect/android/support/rules/RetryOnDeviceErrorRule.kt

```diff
@@ -0,0 +1,23 @@
+package org.odk.collect.android.support.rules
+
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+class RetryOnDeviceErrorRule : TestRule {
+    override fun apply(base: Statement, description: Description): Statement {
+        return object : Statement() {
+            override fun evaluate() {
+                try {
+                    base.evaluate()
+                } catch (e: Throwable) {
+                    if (e::class.simpleName == "RootViewWithoutFocusException") {
```

Comment:
This exception seems to be thrown when the tests can't focus on the app which seems to be a device error that we can't work around. I think always retrying in this scenario is safe as it's unlikely to happen in a genuine fail and if it is from an error in our test logic (trying to assert on the app when the menu drawer is still up), it should happen repeatedly (so we'll still see a fail).

There is a small risk that we have a test where we do something like close the notification drawer and that results in an animation we're not accounting for causing this failure sometimes and that that gets obscured. Again, this wouldn't be an error in the code but a small instability in our test code that's being hidden which I think I can live with.